### PR TITLE
Fancy LED stuff

### PR DESCRIPTION
**Pull request by mlyle**
* Robustness improvements to ws2811 driver
* SystemMod runs faster on RGB LED targets
* Functionality to blend throttle, accessory desired, etc. between colors
* Blending OK in HSV or RGB spaces
* Can also set other LEDs to a default color
* Functionality to direct alarms to subset of LEDs

No UI.

### CODE REVIEW

**Comment by mlyle** [comment]
spurious fail
jenkins, test this please

File: flight/Modules/System/rgbleds.c [hunk 1]
```diff
@@ -0,0 +1,474 @@
...
+		return 0;
+	}
+
+	return float_to_u16(accessory.AccessoryVal);
+}
+
+// The well known 3rd order expansion of sine; 2^15 units/circle
+// Based on http://www.coranac.com/2009/07/sines/
+// Output is Q12
+static int16_t sin_approx(int32_t x)
```
Comment (tracernz) [inline flight/Modules/System/rgbleds.c]: Good fit for https://github.com/d-ronin/dRonin/blob/b323ecce4bc16b80aa638fdff07793a872d2e3f7/flight/Libraries/math/misc_math.h ?

**Review by tracernz — APPROVED** [review]
Looks reasonable. Unfortunate that the Airbot board uses a different pin, results in unnecessarily confusing settings for people with a real Revo (who haven't heard of "OmniBus".

**Comment by tracernz** [comment]
@mlyle Beep.

**Comment by mlyle** [comment]
Blindly moved this.  Can't (easily) test the build on my laptop because I need to worldmagmodel stuff to build...

**Comment by mlyle** [comment]
(After this merges should open an issue that BrainRE1 needs to migrate to new LED support...)

**Review by tracernz — APPROVED** [review]

**Pull request merged**